jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -68,13 +68,22 @@ YUI.add('juju-env-bakery', function(Y) {
         } else {
           this.visitMethod = this._defaultVisitMethod;
         }
-        this.macaroonName = 'Macaroons-' + cfg.serviceName;
+        if (cfg.existingCookie) {
```

Comment:
I'm worried that the logic around this `existingCookie` is going to be error prone. As I'm reading it here its saying to 'use this cookie instead of your default' But there aren't any comments to why you'd want to do this. Could you add some code comments here outlining why this is here.